app/src/main/java/de/danoeh/antennapod/adapter/SelectableAdapter.java

```diff
@@ -0,0 +1,237 @@
+package de.danoeh.antennapod.adapter;
+
+import android.app.Activity;
+import android.util.AttributeSet;
+import android.util.SparseBooleanArray;
+import android.view.ActionMode;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.ViewGroup;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import de.danoeh.antennapod.R;
+
+/**
+ * Used by Recyclerviews that need to provide ability to select items
+ * @param <T>
+ */
+class SelectableAdapter<T extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<T>{
+    private int selectedCount;
+
+    public SelectableAdapter(Activity activity) {
+        this.activity = activity;
+    }
+    @NonNull
+    @Override
+    public T onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        return null;
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull T holder, int position) {
+    }
+
+    @Override
+    public int getItemCount() {
+        return 0;
+    }
+
+    private ActionMode actionMode;
+    private SparseBooleanArray selectedItemPositions = new SparseBooleanArray();
+    private Activity activity;
+
+    public void startActionMode(int pos) {
+        if (inActionMode()) {
+            finish();
```

Comment:
Could be misleading because finish in activities closes the page. How about stopActionMode?